routers/rooms.js

```diff
@@ -499,6 +553,211 @@ router.post('/room/:id/content_flags', authenticateToken, requiresRoomPermission
+        });
+
+        await collection.updateOne(
+            {
+                _id: {
+                    $exists: true,
+                    $eq: room_id
+                }
+            },
+            {
+                $set: {
+                    "userPermissions": {},
+                    "rolePermissions.everyone.viewAndJoin": false,
+                    "rolePermissions.everyone.managePermissions": false,
+                    "description": "This room has been suspended by the Compensation Social moderation team for possible violations of our community standards. Please see https://compensationvr.tk/about/suspension for more information.",
+                }
+            }
+        );
+
+        roomAuditLog(room_id, req.user.id, {
+            'type': AuditEventType.RoomSuspendedByDeveloper,
+            'previous_value': null,
+            'new_value': null,
+            'note': note
+        });
+
+        auditLog(`!! MODERATION ACTION !! - User ${req.user.id} **suspended** room ${room_id}`);
+
+        const players = db.collection("accounts");
+
+        /** @type {string} */
+        var roomname = room.name;
+
+        while (roomname.indexOf("</noparse>") >= 0) {
+            roomname = roomname.replace("</noparse>", "<\\\\noparse>");
+        }
+
+        await players.updateOne(
+            {
+                _id: {
+                    $eq: room.creator_id,
+                    $exists: true
+                }
+            },
+            {
+                $push: {
+                    "notifications": {
+                        template: "room_suspension_notice",
+                        data: {
+                            "headerText": "<smallcaps><color=red>Urgent Moderation Notice",
+                            "bodyText": `We regret to inform you that your room <noparse>"${roomname}"</noparse> has been <color=yellow>suspended</color> by the Compensation Social moderation team.
+
+                            For more information, please see
+                            <color=#FF5566>https://compensationvr.tk/about/suspension</color>`
+                        }
+                    }
+                }
+            }
+        );
+
+        var send = WebSocketV2_MessageTemplate;
+        send.code = "urgent_notification_recieved";
+        send.data = {};
+        require('./ws/WebSocketServerV2').ws_connected_clients[room.creator_id]?.socket?.send(JSON.stringify(send, null, 5));
+
+        return res.status(200).json({
+            code: "success",
+            message: "The operation was successful."
+        });
+    } catch (ex) {
+        res.status(500).json({
+            code: "internal_error",
+            message: "An internal error occurred and we could not suspend that room."
+        });
+        throw ex;
+    }
+});
+
+router.post("/room/:id/moderation-terminate", authenticateDeveloperToken, async (req, res) => {
```

Comment:
* https://api.compensationvr.tk/api/rooms/room/:id/moderation-terminate
* Terminates the specified room with a note.
* Usable only by developers.